pypasta/Config.py

```diff
@@ -299,6 +304,28 @@ def load_upstream_hashes(self, force_reload=False):
 
         self.upstream_hashes = upstream
 
+    def pull_patchwork_patches(self):
+        since = self.repo.mbox.latest_message_date()
+        index = self.repo.mbox.mbox_raw.index
+        log.info('Pulling patches from patchwork created since %s', since)
+        lists = dict(self.mbox_raw)
+        pulled = skipped = 0
+        try:
+            for patch, msg_id, list_id in self.pw.pull_patches(since):
+                if list_id not in lists or msg_id in index:
+                    skipped += 1
+                    continue
+                elif isinstance(lists[list_id], str):
+                    # Replace path str with file object
+                    lists[list_id] = open(lists[list_id], 'a')
+                lists[list_id].write(patch + '\n')
+                pulled += 1
+        finally:
+            for f_mbox in [file for file in lists.values()
+                           if not isinstance(file, str)]:
```

Comment:
Can we somehow avoid the isinstance dance? isinstance always makes the assumption that the datatype might change. This is error prone.

Reply:
The datatype is changing here. What is happening here is that we append a patch to its corresponding list after its download. The path to the list file is replaced with an actual file object then.
We can still avoid using isinstance of course.